sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -574,6 +573,14 @@ object PartitioningUtils extends SQLConfHelper {
     }
   }
 
+  /**
+   * Checks whether a given data type can be used as a partition column.
+   */
+  def canPartitionOn(dateType: DataType): Boolean = dateType match {
+    case st: StringType => st.isDefaultCollation
```

Comment:
Hey, I know that Spark codebase is not big on comments, but maybe we should add a comment why partitioning is disabled + jira ticket to enable the partitioning :) ?

Reply:
added the comment but not sure what you meant regarding the jira ticket to enable the partitioning - there is no plan to do this AFAIK and also I don't see how it would even be possible unless we revamp the entire partitioning logic - ie. not write the actual value to the file system